packages/mermaid/src/diagrams/class/classRenderer-v2.ts

```diff
@@ -230,7 +230,7 @@ export const addRelations = function (relations: ClassRelation[], g: graphlib.Gr
       //Set relationship style and line type
       classes: 'relation',
       pattern: edge.relation.lineType == 1 ? 'dashed' : 'solid',
-      id: 'id' + cnt,
+      id: 'id_' + edge.id1 + '_' + edge.id2,
```

Comment:
Is this the only relevant part for lexical IDs?
If so, can you reset your branch to develop and only have this in the PR?

Reply:
I think the cnt should also be in the ID. 

Otherwise, id of [a, b_c] and [a_b, c] will both be id_a_b_c, leading to a conflict.